Gtk: generate events for unknown keys

martinetd opened the pull request:
Follow up (side-follow?) for #2575, quoting the main commit message:

```
gtk/window: generate WXK_NONE key event for unknown keys

There are still plenty of X11 keys which have no WXK mapping and might be useful
for applications to act upon.

Instead of trying to remap all of them, generate a WXK_NONE key event and let
applications deal with it using GetRawKeyCode() if appropriate.
```


Tested with xdotool but I actualy have plenty of fn+keys that would allow testing it on keyboard sample... Comparing a key with binding (XF86LaunchA) and one without (XF86Battery):
```
$ sleep 1; xdotool key XF86LaunchA; xdotool key XF86Battery

   Hook     LAUNCH_APP1   432   ----    0 (U+0000)  269025098    0x00000080  (  217,  619)  No
KeyDown     LAUNCH_APP1   432   ----    0 (U+0000)  269025098    0x00000080  (  217,  619)  No
   Char     LAUNCH_APP1   432   ----    0 (U+0000)  269025098    0x00000080  (  217,  619)  No
  KeyUp     LAUNCH_APP1   432   ----    0 (U+0000)  269025098    0x00000080  (  217,  619)  Yes
   Hook         unknown     0   ----    0 (U+0000)  269025171    0x000000f4  (  470,  651)  No
KeyDown         unknown     0   ----    0 (U+0000)  269025171    0x000000f4  (  470,  651)  No
  KeyUp         unknown     0   ----    0 (U+0000)  269025171    0x000000f4  (  470,  651)  Yes
```

Note that in particular there is no Char event for WXK_NONE, which totally makes sense as there is no associated char -- conversely I'd be tempted to nitpick and say LAUNCH_APP1 does not deserve one either?

269025171 in this case is comparable with either the gdk or the x11 symbol, I have it in a few headers on my system (in all cases defined as hex, 0x1008ff93 = 269025171)
```
/usr/include/gtk-3.0/gdk/gdkkeysyms.h
2269:#define GDK_KEY_Battery 0x1008ff93

/usr/include/gtk-3.0/gdk/gdkkeysyms-compat.h
2260:#define GDK_Battery 0x1008ff93

/usr/include/xkbcommon/xkbcommon-keysyms.h
2618:#define XKB_KEY_XF86Battery		0x1008FF93   /* Display battery information */

/usr/include/X11/XF86keysym.h
170:#define XF86XK_Battery		0x1008FF93   /* Display battery information */
```

martinetd committed:
gtk/window: generate WXK_NONE key event for unknown keys

There are still plenty of X11 keys which have no WXK mapping and might be useful
for applications to act upon.

Instead of trying to remap all of them, generate a WXK_NONE key event and let
applications deal with it using GetRawKeyCode() if appropriate.

martinetd committed:
gtk/window: cleanup dead code around wxTranslateGTKKeyEventToWx

The previous commit made wxTranslateGTKKeyEventToWx never fail, so make it void
and remove all dead code resulting from the lack of failure.

vadz commented:
Thanks, this looks good to me, but we need to update the documentation to mention that key events with `WXK_NONE` are possible now.

@paulcor Do you have any objections to merging this?

martinetd commented:
> Thanks, this looks good to me, but we need to update the documentation to mention that key events with WXK_NONE are possible now.

Right, I'm always bad with that... Documentation of wxKeyEvent in interface/wx/event.h? Is there anything else you can think of?
If that's it I'll update it a bit to say if both GetUnicodeKey and GetKeyCode returned WXK_NONE then GetRawKeyCode can be used to distinguish keys but is platform specific.

vadz commented:
> Documentation of wxKeyEvent in interface/wx/event.h?

Yes, exactly.

> Is there anything else you can think of? If that's it I'll update it a bit to say if both GetUnicodeKey and GetKeyCode returned WXK_NONE then GetRawKeyCode can be used to distinguish keys but is platform specific.

I think this would be quite enough, TIA!

martinetd committed:
Document generation of WXK_NONE key events for unknown keys

Add a paragraph for keys where GetUnicodeKey and GetKeyCode both return
WXK_NONE, that GetRawKeyCode is platform specific, and that such keys
generate no wxEVT_CHAR event.

vadz commented:
Perfect, thanks! Will merge in a moment.
